packages/server/src/games.ts

```diff
@@ -58,3 +58,99 @@ export function updateGameState(io: IOServer, store: Store, roomName: string, ga
     io.in(roomName).emit(GameEvent.CurrentState, updatedGameState);
   }
 }
+
+/**
+ * Calculate the immediate new game state, as a result of a player's action. This is not necessarily the final state
+ * but an intermediate one to help the following calculations.
+ */
+function calculatePlayerAction(oldState: GameState, playerAction: PlayerAction) {
+  const activePlayer = oldState.players.filter((playerState) => playerState.username === playerAction.playerName)[0];
+  // TODO: Later on we will need to add order of players, built into the model
```

Comment:
yes because I want to make the version with 4 just after we'll have this working